HeterogeneousCore/SonicCore/src/SonicClientBase.cc

```diff
@@ -12,6 +19,18 @@ SonicClientBase::SonicClientBase(const edm::ParameterSet& params,
   if (!clientName_.empty())
     fullDebugName_ += ":" + clientName_;
 
+  std::vector<edm::ParameterSet> retryPSetList = params.getParameter<std::vector<edm::ParameterSet>>("Retry");
+
+  for (const auto& retryPSet : retryPSetList) {
+        std::string actionType = retryPSet.getParameter<std::string>("retryType");
+
+        auto retryAction = RetryActionFactory::get()->create(actionType, retryPSet, this);
+        if (retryAction) {
+            //Convert to  RetryActionPtr Type from raw pointer of retryAction 
+            retryActions_.emplace_back(RetryActionPtr(retryAction.release()));
+        }
```

Comment:
add an `else` here to throw an exception if `retryAction` is null, since that means the `retryType` was not known